storage_server/report_server.py

```diff
@@ -184,30 +208,49 @@ def replaceConfigInfo(self, run_id, config_values):
         return True
 
     @decorators.catch_sqlalchemy
-    def addBuildAction(self, run_id, build_cmd, check_cmd):
+    def addBuildAction(self,
+                       run_id,
+                       build_cmd,
+                       check_cmd,
+                       analyzer_type,
+                       analyzed_source_file):
         '''
         '''
         try:
+
             build_actions = \
                 self.session.query(BuildAction) \
-                            .filter(and_(BuildAction.run_id == run_id,
-                                         BuildAction.build_cmd == build_cmd,
-                                         BuildAction.check_cmd == check_cmd))\
-                            .all()
+                    .filter(and_(BuildAction.run_id == run_id,
+                                BuildAction.build_cmd == build_cmd,
```

Comment:
It may cause problems if we include build command in this filter. What if, between two runs I change the compilation parameters for a file and modify that file. In that case, the old results will remain for that file. I think, we should delete all reports to a source file if we reanalize that source file.